src/shaka_scheme/system/lexer/Tokenizer.hpp

```diff
@@ -0,0 +1,1011 @@
+          while (is_subsequent(in.peek())) {
+            buffer += in.get();
+          }
+          return Token(Token::Type::IDENTIFIER, buffer);
+          // . <dot subsequent> <subsequent>*
+        } else if (in.peek() == '.') {
+          buffer += in.get();
+          if (is_dot_subsequent(in.peek())) {
+            buffer += in.get();
+            while (is_subsequent(in.peek())) {
+              buffer += in.get();
+            }
+            return Token(Token::Type::IDENTIFIER, buffer);
+
+            // if it's a digit, it's probably a number
+          } else if (std::isdigit(in.peek())) {
+            return parse_number(buffer);
+
+            // no <dot subsequent> ==> error!
+          } else {
+            throw shaka::TokenizerException(20014, "Tokenizer.parse_token: bad "
+                "dot subsequent identifier");
+            return Token(Token::Type::END_OF_FILE);
+          }
+
+          // Parse in a number
+        } else if (std::isdigit(in.peek())) {
+          std::string buffer;
+          return parse_number(buffer);
+
+          // No other identifier? Just <explicit sign> is fine.
+        } else {
+          return Token(Token::Type::IDENTIFIER, buffer);
+        }
+
+        // Identifier ==> . <dot subsequent> <subsequent>
+      } else if (in.peek() == '.') {
+        std::string buffer;
+        buffer += in.get();
+        if (is_dot_subsequent(in.peek())) {
+          buffer += in.get();
+          while (is_subsequent(in.peek())) {
+            buffer += in.get();
+          }
+          return Token(Token::Type::IDENTIFIER, buffer);
+        } else {
+          throw shaka::TokenizerException(20015, "Tokenizer.parse_token: bad "
+              "dot subsequent identifier");
+          return Token(Token::Type::END_OF_FILE);
+        }
+
+        // <number> ==> <num 10> for now
+        /// @todo Add real, rational, complex for different radixes.
+      } else if (std::isdigit(in.peek())) {
+        std::string buffer;
+        return parse_number(buffer);
+
+        // NOT A VALID TOKEN!!!
+      } else {
+        if (DEBUG_PRINT) {
+          std::cerr << "BROKE ON: " << in.peek() << std::endl;
+        }
+        throw shaka::TokenizerException(20016, "Tokenizer.parse_token: invalid "
+            "token");
+        return Token(Token::Type::END_OF_FILE);
+      }
+    }
+    throw 1;
+    return Token(Token::Type::END_OF_FILE);
+  }
+
+  bool is_delimiter(char c) {
+    return (
+        std::isspace(c)
+            || c == '|'
+            || c == '('
+            || c == ')'
+            || c == '\"'
+            || c == ';'
+    );
+  }
+
+  bool is_special_initial(char c) {
+    return (
+        c == '!' ||
+            c == '$' ||
+            c == '%' ||
+            c == '&' ||
+            c == '*' ||
+            c == '/' ||
+            c == '!' ||
+            c == ':' ||
+            c == '>' ||
+            c == '=' ||
+            c == '>' ||
+            c == '?' ||
```

Comment:
There is a small bug here, you check for the presence of a `>` symbol twice instead of checking for both `>` and `<`. Minor detail. I actually thought I had fixed that at the beginning of last summer, but we can always change it later